flink-streaming-connectors/flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisContainer.java

```diff
@@ -0,0 +1,268 @@
+	 */
+	@Override
+	public void publish(final String channelName, final String message) {
+		Jedis jedis = null;
+		try {
+			jedis = getInstance();
+			jedis.publish(channelName, message);
+		} catch (Exception e) {
+			if (LOG.isErrorEnabled()) {
+				LOG.error("Cannot send Redis message with command PUBLISH to channel {} error message {}",
+					channelName, e.getMessage());
+			}
+		} finally {
+			returnInstance(jedis);
+		}
+	}
+
+	/**
+	 * Set key to hold the string value. If key already holds a value, it is overwritten,
+	 * regardless of its type. Any previous time to live associated with the key is
+	 * discarded on successful SET operation.
+	 *
+	 * @param key   the key name in which value to be set
+	 * @param value the value
+	 */
+	@Override
+	public void set(final String key, final String value) {
+		Jedis jedis = null;
+		try {
+			jedis = getInstance();
+			jedis.set(key, value);
+		} catch (Exception e) {
+			if (LOG.isErrorEnabled()) {
+				LOG.error("Cannot send Redis message with command SET to key {} error message {}",
+					key, e.getMessage());
+			}
+		} finally {
+			returnInstance(jedis);
+		}
+	}
+
+	/**
+	 * Adds all the element arguments to the HyperLogLog data structure
+	 * stored at the variable name specified as first argument.
+	 *
+	 * @param key     The name of the key
+	 * @param element the element
+	 */
+	@Override
+	public void pfadd(final String key, final String element) {
+		Jedis jedis = null;
+		try {
+			jedis = getInstance();
+			jedis.pfadd(key, element);
+		} catch (Exception e) {
+			if (LOG.isErrorEnabled()) {
+				LOG.error("Cannot send Redis message with command PFADD to key {} error message {}",
+					key, e.getMessage());
+			}
+		} finally {
+			returnInstance(jedis);
+		}
+	}
+
+	/**
+	 * Adds the specified member with the specified scores to the sorted set stored at key
+	 *
+	 * @param setName The name of the Sorted Set
+	 * @param element element to be added
+	 * @param score   Score of the element
+	 */
+	@Override
+	public void zadd(final String setName, final String element, final String score) {
+		Jedis jedis = null;
+		try {
+			jedis = getInstance();
+			jedis.zadd(setName, Double.valueOf(score), element);
+		} catch (Exception e) {
+			if (LOG.isErrorEnabled()) {
+				LOG.error("Cannot send Redis message with command ZADD to set {} error message {}",
+					setName, e.getMessage());
+			}
+		} finally {
+			returnInstance(jedis);
+		}
+	}
+
+	/**
+	 * Returns Jedis instance from the pool.
+	 *
+	 * @return the Jedis instance
+     */
+	private Jedis getInstance() {
+		if (jedisSentinelPool != null) {
+			return jedisSentinelPool.getResource();
+		} else if (jedisPool != null) {
+			return jedisPool.getResource();
+		} else {
+			throw new IllegalArgumentException("Jedis Pool not found");
+		}
+	}
+
+	/**
+	 * Closes the jedis instance after finishing the command.
+	 *
+	 * @param jedis The jedis instance
+     */
+	private void returnInstance(final Jedis jedis) {
```

Comment:
Maybe "releaseInstance" instead of "returnInstance" (return sounds a little bit like "get" -- but is not a getter)